exonum/src/storage/sparse_list_index.rs

```diff
@@ -0,0 +1,838 @@
+    ///
+    /// index.extend([1, 2, 3, 4, 5].iter().cloned());
+    /// index.remove(3);
+    ///
+    /// for val in index.iter_from(3) {
+    ///     println!("{:?}", val);
+    /// }
+    /// ```
+    pub fn iter_from(&self, from: u64) -> SparseListIndexIter<V> {
+        SparseListIndexIter { base_iter: self.base.iter_from(&(), &from) }
+    }
+}
+
+
+impl<'a, V> SparseListIndex<&'a mut Fork, V>
+where
+    V: StorageValue,
+{
+    fn set_size(&mut self, size: SparseListSize) {
+        self.base.put(&(), size);
+        self.size.set(Some(size));
+    }
+
+    /// Appends an element to the back of the list.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use exonum::storage::{MemoryDB, Database, SparseListIndex};
+    ///
+    /// let db = MemoryDB::new();
+    /// let mut fork = db.fork();
+    /// let mut index = SparseListIndex::new(vec![1, 2, 3], &mut fork);
+    ///
+    /// index.push(1);
+    /// assert!(!index.is_empty());
+    /// ```
+    pub fn push(&mut self, value: V) {
+        let mut size = self.size();
+        self.base.put(&size.capacity, value);
+        size.capacity += 1;
+        size.length += 1;
+        self.set_size(size);
+    }
+
+    /// Removes the last element from the list and returns it, or None if it is empty.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use exonum::storage::{MemoryDB, Database, SparseListIndex};
+    ///
+    /// let db = MemoryDB::new();
+    /// let mut fork = db.fork();
+    /// let mut index = SparseListIndex::new(vec![1, 2, 3], &mut fork);
+    /// assert_eq!(None, index.pop());
+    ///
+    /// index.push(1);
+    /// assert_eq!(Some(1), index.pop());
+    /// ```
+    pub fn pop(&mut self) -> Option<V> {
+        let mut size = self.size();
+        match size.length {
+            0 => None,
+            l => {
+                let v = self.base.get(&(l - 1));
+                self.base.remove(&(l - 1));
+                size.length -= 1;
+                self.set_size(size);
+                v
+            }
+        }
+    }
+
+    /// Removes the element with the given index from the list and returns it,
+    /// or None if it is empty.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use exonum::storage::{MemoryDB, Database, SparseListIndex};
+    ///
+    /// let db = MemoryDB::new();
+    /// let mut fork = db.fork();
+    /// let mut index = SparseListIndex::new(vec![1, 2, 3], &mut fork);
+    /// assert_eq!(0, index.capacity());
+    ///
+    /// index.push(10);
+    /// index.push(12);
+    ///
+    /// assert_eq!(Some(10), index.remove(0));
+    /// assert_eq!(None, index.remove(0));
+    /// assert_eq!(2, index.capacity());
+    /// assert_eq!(1, index.len());
+
+    /// assert_eq!(Some(12), index.remove(1));
+    /// assert_eq!(2, index.capacity());
+    /// ```
+    pub fn remove(&mut self, index: u64) -> Option<V> {
+        let mut size = self.size();
+        if index >= size.capacity {
+            warn!(
```

Comment:
I think the `warning` is unnecessary here. In some situations, an attempt to remove outside the capacity can be expected behavior. Moreover, in this case, it would be necessary to add a similar warning for `get` method for consistency.